app/javascript/packs/base.jsx

```diff
@@ -9,6 +9,14 @@ import { trackCreateAccountClicks } from '@utilities/ahoy/trackEvents';
 import { showWindowModal, closeWindowModal } from '@utilities/showModal';
 import * as Runtime from '@utilities/runtime';
 
+Document.prototype.ready = new Promise((resolve) => {
```

Comment:
We identified in a previous PR that this may be useful to pull out into a helper as it appears in a couple of places but I wasn't a 100% sure how can be done because it extends the document.